lib/view_component_reflex/state_adapter/redis.rb

```diff
@@ -0,0 +1,74 @@
+# A redis based adapter for component_reflex
+#
+# ViewComponentReflex::Engine.configure do |config|
+#   config.state_adapter = ViewComponentReflex::StateAdapter::Redis.new(
+#       redis_opts: {
+#           url: "redis://localhost:6379/1", driver: :hiredis
+#       },
+#       ttl: 3600)
+# end
+
+module ViewComponentReflex
+  module StateAdapter
+    class Redis
+      attr_reader :client, :ttl
+
+      def initialize(redis_opts:, ttl: 3600)
+        @client = Redis.new(redis_opts)
+        @ttl = ttl
+      end
+
+      def state(request, key)
+        cache_key = get_key(request, key)
+        value = client.hgetall(cache_key)
+
+        return {} if value.nil?
+
+        value.map do |k, v|
+          [k, JSON.parse(v)]
```

Comment:
This is a sister comment to the `to_json` one

```suggestion
          [k, Marshal.load(v)]
```